### ISSUE TITLE
Question about Pre-training Olmo 7B

### ISSUE THREAD

**Issue by michaellin99999** [allenai/OLMo#497]
### ❓ The question

We tried pretraining Olmo 7B using the script provided and the 7TB data provided. 
We are now at steps 5000, with a perplexity around 20. currently the learning rate is 3 x E-4
and we notice the loss is actually becoming flat.
Is this a queue to decrease the learning rate to 3 x E-5?  
another question. We are using HGX x 8 H100 to pretrain. How long will this typically take and are there any tips we can follow to successfully pretrain and reproduce the steps of pretraining it?   To reproduce the results, does perplexity need to reach below 8?

![clipboard-image](https://github.com/allenai/OLMo/assets/57974973/9126d8eb-e7b8-4289-9bb1-e79beec5d26d)

**Comment by 2015aroras** [comment]
Regarding learning rate, if you used our 7B training config then the first 5k steps were warming up the LR linearly from 3e-5 to 3e-4. If you continue training, it should start linearly decaying the LR back to 3e-4 over the next ~500k steps. So, you might not need to worry about tweaking the LR yourself.

In terms of how long it will take, you can probably get a good estimate by taking the training time of the first 5k steps and multiplying by 100 (we trained ~500k steps). Regarding reproducing our results, it may be good to look at our results on our evaluation suite over our training. You can roughly convert tokens to steps by dividing by 4M.

<img width="540" alt="Accuracy score progression of OLMo-7B on 8 core end-tasks score from Catwalk evaluation suite" src="https://github.com/allenai/OLMo/assets/19700980/929a31c5-dc93-47f4-ba30-486abaf41656">

**Comment by dumitrac** [comment]
@michaellin99999 - please reopen this item or a new one if you have followup questions.
